com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest.ordersOfCustomers [HZ-1310]

_master_ (commit be07ca930186e8140f8c5fb8e6afbe60a7cadc92)

Failed on Oracle JDK 8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK8-nightly/904/testReport/com.hazelcast.jet.cdc.postgres/MultiTableCacheIntegrationTest/ordersOfCustomers/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<{1001=Customer: Customer {id=1001, firstName=Sally, lastName=Thomas, email=sally.thomas@acme.com}, Orders: {10001=Order {orderNumber=10001, orderDate=Sat Jan 16 00:00:00 UTC 2016, purchaser=1001, quantity=1, productId=102}}, 1002=Customer: Customer {id=1002, firstName=George, lastName=Bailey, email=gbailey@foobar.com}, Orders: {10002=Order {orderNumber=10002, orderDate=Sun Jan 17 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=105}, 10003=Order {orderNumber=10003, orderDate=Fri Feb 19 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=106}}, 1003=Customer: Customer {id=1003, firstName=Edward, lastName=Walker, email=ed@walker.com}, Orders: {10004=Order {orderNumber=10004, orderDate=Sun Feb 21 00:00:00 UTC 2016, purchaser=1003, quantity=1000, productId=107}}, 1004=Customer: Customer {id=1004, firstName=Anne1000, lastName=Kretchmar, email=annek@noanswer.org}, Orders: {}, 1005=Customer: null, Orders: {}}> but was:<{1001=Customer: Customer {id=1001, firstName=Sally, lastName=Thomas, email=sally.thomas@acme.com}, Orders: {10001=Order {orderNumber=10001, orderDate=Sat Jan 16 00:00:00 UTC 2016, purchaser=1001, quantity=1, productId=102}}, 1002=Customer: Customer {id=1002, firstName=George, lastName=Bailey, email=gbailey@foobar.com}, Orders: {10002=Order {orderNumber=10002, orderDate=Sun Jan 17 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=105}, 10003=Order {orderNumber=10003, orderDate=Fri Feb 19 00:00:00 UTC 2016, purchaser=1002, quantity=2, productId=106}}, 1003=Customer: Customer {id=1003, firstName=Edward, lastName=Walker, email=ed@walker.com}, Orders: {10004=Order {orderNumber=10004, orderDate=Sun Feb 21 00:00:00 UTC 2016, purchaser=1003, quantity=1000, productId=107}}, 1004=Customer: Customer {id=1004, firstName=Anne1000, lastName=Kretchmar, email=annek@noanswer.org}, Orders: {}}>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at com.hazelcast.test.HazelcastTestSupport.lambda$assertEqualsEventually$8(HazelcastTestSupport.java:1005)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1226)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1328)
	at com.hazelcast.test.HazelcastTestSupport.assertEqualsEventually(HazelcastTestSupport.java:1005)
	at com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest.ordersOfCustomers(MultiTableCacheIntegrationTest.java:131)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
07:03:34,087 DEBUG |ordersOfCustomers| - [PrefixingImageNameSubstitutor] Time-limited test - No prefix is configured
07:03:34,087 DEBUG |ordersOfCustomers| - [ImageNameSubstitutor] Time-limited test - Did not find a substitute image for debezium/example-postgres:1.7 (using image substitutor: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor'))
07:03:34,087 DEBUG |ordersOfCustomers| - [AbstractImagePullPolicy] Time-limited test - Using locally available and not pulling image: debezium/example-postgres:1.7
07:03:34,087  INFO |ordersOfCustomers| - [7]] Time-limited test - Creating container for image: debezium/example-postgres:1.7
07:03:34,087 DEBUG |ordersOfCustomers| - [RegistryAuthLocator] Time-limited test - Looking up auth config for image: debezium/example-postgres:1.7 at registry: index.docker.io
07:03:34,087 DEBUG |ordersOfCustomers| - [RegistryAuthLocator] Time-limited test - No matching Auth Configs - falling back to defaultAuthConfig [null]
07:03:34,087 DEBUG |ordersOfCustomers| - [AuthDelegatingDockerClientConfig] Time-limited test - Effective auth config [null]
07:03:34,590  INFO |ordersOfCustomers| - [7]] Time-limited test - Container debezium/example-postgres:1.7 is starting: 03464cdbe638d5194cad9748bef8fabd2393b87a725d90075e57fae67c2847b5
07:03:35,346 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:35,346 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: This user must also own the server process.
07:03:35,346 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The files belonging to this database system will be owned by user "postgres".
07:03:35,356 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: creating subdirectories ... ok
07:03:35,356 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: fixing permissions on existing directory /var/lib/postgresql/data ... ok
07:03:35,356 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:35,356 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: Data page checksums are disabled.
07:03:35,356 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:35,356 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The default text search configuration will be set to "english".
07:03:35,357 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The default database encoding has accordingly been set to "UTF8".
07:03:35,357 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: The database cluster will be initialized with locale "en_US.utf8".
07:03:35,367 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting default max_connections ... 100
07:03:35,388 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting default shared_buffers ... 128MB
07:03:35,422 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting dynamic shared memory implementation ... posix
07:03:35,422 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: selecting default timezone ... Etc/UTC
07:03:35,432 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: creating configuration files ... ok
07:03:35,578 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: running bootstrap script ... ok
07:03:36,053 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: performing post-bootstrap initialization ... ok
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT:     pg_ctl -D /var/lib/postgresql/data -l logfile start
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: --auth-local and --auth-host, the next time you run initdb.
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: You can change this by editing pg_hba.conf or using the option -A, or
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: WARNING: enabling "trust" authentication for local connections
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: Success. You can now start the database server using:
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:36,161 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: syncing data to disk ... ok
07:03:36,199 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: waiting for server to start....2022-07-06 07:03:36.198 GMT [49] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
07:03:36,209 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-07-06 07:03:36.205 GMT [50] LOG:  database system was shut down at 2022-07-06 07:03:36 GMT
07:03:36,219 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-07-06 07:03:36.210 GMT [49] LOG:  database system is ready to accept connections
07:03:36,287 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: server started
07:03:36,287 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT:  done
07:03:36,382 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: /usr/local/bin/docker-entrypoint.sh: sourcing /docker-entrypoint-initdb.d/init-permissions.sh
07:03:36,382 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:36,392 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/inventory.sql
07:03:36,392 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:36,432 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: SET
07:03:36,432 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE SCHEMA
07:03:37,408 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE EXTENSION
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 1
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 9
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER SEQUENCE
07:03:37,418 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:03:37,428 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:37,428 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:37,428 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 3
07:03:37,428 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:03:37,428 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 4
07:03:37,429 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:03:37,429 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER SEQUENCE
07:03:37,429 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:03:37,429 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: INSERT 0 4
07:03:37,429 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER TABLE
07:03:37,429 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: ALTER SEQUENCE
07:03:37,429 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: CREATE TABLE
07:03:37,439 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-07-06 07:03:37.432 GMT [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
07:03:37,439 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-07-06 07:03:37.429 GMT [76] FATAL:  terminating autovacuum process due to administrator command
07:03:37,439 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: .2022-07-06 07:03:37.429 GMT [49] LOG:  aborting any active transactions
07:03:37,439 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: waiting for server to shut down...2022-07-06 07:03:37.429 GMT [49] LOG:  received fast shutdown request
07:03:37,449 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-07-06 07:03:37.449 GMT [51] LOG:  shutting down
07:03:37,469 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 2022-07-06 07:03:37.468 GMT [49] LOG:  database system is shut down
07:03:37,530 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:37,530 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: PostgreSQL init process complete; ready for start up.
07:03:37,530 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: 
07:03:37,530 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT: server stopped
07:03:37,530 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDOUT:  done
07:03:37,549 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-07-06 07:03:37.549 GMT [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
07:03:37,549 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-07-06 07:03:37.549 GMT [1] LOG:  listening on IPv6 address "::", port 5432
07:03:37,549 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-07-06 07:03:37.549 GMT [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
07:03:37,562 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-07-06 07:03:37.561 GMT [78] LOG:  database system was shut down at 2022-07-06 07:03:37 GMT
07:03:37,572 DEBUG |ordersOfCustomers| - [WaitingConsumer] Time-limited test - STDERR: 2022-07-06 07:03:37.567 GMT [1] LOG:  database system is ready to accept connections
07:03:37,572  INFO |ordersOfCustomers| - [7]] Time-limited test - Container debezium/example-postgres:1.7 started in PT3.486S
07:03:37,586  INFO |ordersOfCustomers| - [MetricsConfigHelper] ordersOfCustomers - [LOCAL] [dev] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
07:03:37,587  INFO |ordersOfCustomers| - [logo] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
07:03:37,587  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
07:03:37,587  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220706 - be07ca9) starting at [127.0.0.1]:5701
07:03:37,587  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Cluster name: dev
07:03:37,587  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
07:03:37,587  INFO |ordersOfCustomers| - [system] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Jet is enabled
07:03:37,592  INFO |ordersOfCustomers| - [MetricsConfigHelper] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
07:03:37,598  WARN |ordersOfCustomers| - [CPSubsystem] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
07:03:37,602  INFO |ordersOfCustomers| - [JetServiceBackend] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
07:03:37,602  INFO |ordersOfCustomers| - [Diagnostics] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
07:03:37,603  INFO |ordersOfCustomers| - [LifecycleService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTING
07:03:37,603  INFO |ordersOfCustomers| - [ClusterService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - c31a8bd3-6585-40ed-8a46-189e622d01ce this
]

07:03:37,603  INFO |ordersOfCustomers| - [JobCoordinationService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Jet started scanning for jobs
07:03:37,603  INFO |ordersOfCustomers| - [LifecycleService] ordersOfCustomers - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTED
07:03:37,606  INFO |ordersOfCustomers| - [PartitionStateManager] hz.heuristic_neumann.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initializing cluster partition table arrangement...
07:03:37,614  INFO |ordersOfCustomers| - [JobCoordinationService] hz.heuristic_neumann.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Starting job 0848-d36e-e0c0-0001 based on submit request
07:03:37,617  INFO |ordersOfCustomers| - [MasterJobContext] hz.heuristic_neumann.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Didn't find any snapshot to restore for job '0848-d36e-e0c0-0001', execution 0848-d36e-e0c1-0001
07:03:37,617  INFO |ordersOfCustomers| - [MasterJobContext] hz.heuristic_neumann.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Start executing job '0848-d36e-e0c0-0001', execution 0848-d36e-e0c1-0001, execution graph in DOT format:
digraph DAG {
	"source" [localParallelism=1];
	"filter" [localParallelism=2];
	"map-stateful-keyed" [localParallelism=2];
	"mapWithEntryProcessorSink(cache)" [localParallelism=2];
	"filter-2" [localParallelism=2];
	"map-stateful-keyed-2" [localParallelism=2];
	"mapWithEntryProcessorSink(cache)-2" [localParallelism=2];
	"source" -> "filter" [taillabel=0, queueSize=1024];
	"source" -> "filter-2" [taillabel=1, queueSize=1024];
	"filter-2" -> "map-stateful-keyed-2" [label="distributed-partitioned", queueSize=1024];
	"filter" -> "map-stateful-keyed" [label="distributed-partitioned", queueSize=1024];
	"map-stateful-keyed" -> "mapWithEntryProcessorSink(cache)" [label="partitioned", queueSize=1024];
	"map-stateful-keyed-2" -> "mapWithEntryProcessorSink(cache)-2" [label="partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com/ to visualize the printed graph.
07:03:37,621  INFO |ordersOfCustomers| - [JobExecutionService] hz.heuristic_neumann.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution plan for jobId=0848-d36e-e0c0-0001, jobName='0848-d36e-e0c0-0001', executionId=0848-d36e-e0c1-0001 initialized
07:03:37,622  INFO |ordersOfCustomers| - [JobExecutionService] hz.heuristic_neumann.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Start execution of job '0848-d36e-e0c0-0001', execution 0848-d36e-e0c1-0001 from coordinator [127.0.0.1]:5701
07:03:37,623  INFO |ordersOfCustomers| - [ChangeRecordCdcSourceP] hz.heuristic_neumann.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [0848-d36e-e0c0-0001/source#0] source has 'retry strategy' set to 'max attempts = -1, interval function = IntervalFunction (constant of 1000ms)'
07:03:37,623  INFO |ordersOfCustomers| - [ChangeRecordCdcSourceP] hz.heuristic_neumann.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [0848-d36e-e0c0-0001/source#0] source has 'commit period' set to '10000000000'
07:03:37,623  INFO |ordersOfCustomers| - [ChangeRecordCdcSourceP] hz.heuristic_neumann.jet.blocking.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [0848-d36e-e0c0-0001/source#0] source has 'clear state on reconnect' set to 'false'
07:03:37,623  WARN |ordersOfCustomers| - [PostgresConnectorConfig] hz.heuristic_neumann.jet.blocking.thread-0 - Configuration property 'truncate.handling.mode' is deprecated and will be removed in future versions. Please use 'skipped.operations' instead.
07:03:37,623  WARN |ordersOfCustomers| - [Configuration] hz.heuristic_neumann.jet.blocking.thread-0 - Using configuration property "table.whitelist" is deprecated and will be removed in future versions. Please use "table.include.list" instead.
07:03:37,623  WARN |ordersOfCustomers| - [PostgresConnectorConfig] hz.heuristic_neumann.jet.blocking.thread-0 - Configuration property 'toasted.value.placeholder' is deprecated and will be removed in future versions. Please use 'unavailable.value.placeholder' instead.
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 - Starting PostgresConnectorTask with configuration:
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    reconnect.behavior = max attempts = -1, interval function = IntervalFunction (constant of 1000ms)
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    connector.class = io.debezium.connector.postgresql.PostgresConnector
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    db.specific.extra.fields = schema
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    database.user = postgres
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    database.dbname = postgres
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    database.server.name = 58fba57b-f9cf-49cf-9856-aafc50c65fc5
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    sequence.extractor.class = com.hazelcast.jet.cdc.postgres.impl.PostgresSequenceExtractor
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    database.port = 57109
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    table.whitelist = inventory.customers,inventory.orders
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    tombstones.on.delete = false
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    database.hostname = 172.17.0.1
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    database.password = ********
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    name = source
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    snapshot.mode = initial
07:03:37,623  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 -    database.history = com.hazelcast.jet.cdc.impl.CdcSourceP$DatabaseHistoryImpl
07:03:37,634  INFO |ordersOfCustomers| - [JdbcConnection] pool-265-thread-1 - Connection gracefully closed
07:03:37,653  INFO |ordersOfCustomers| - [BaseSourceTask] hz.heuristic_neumann.jet.blocking.thread-0 - No previous offsets found
07:03:37,655  INFO |ordersOfCustomers| - [PostgresConnectorTask] hz.heuristic_neumann.jet.blocking.thread-0 - user 'postgres' connected to database 'postgres' on PostgreSQL 11.14 (Debian 11.14-1.pgdg90+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516, 64-bit with roles:
	role 'pg_read_all_settings' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_stat_scan_tables' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_write_server_files' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_monitor' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_read_server_files' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_execute_server_program' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_read_all_stats' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'pg_signal_backend' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
	role 'postgres' [superuser: true, replication: true, inherit: true, create role: true, create db: true, can log in: true]
07:03:37,656  INFO |ordersOfCustomers| - [PostgresConnection] hz.heuristic_neumann.jet.blocking.thread-0 - Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=null, catalogXmin=null]
07:03:37,656  INFO |ordersOfCustomers| - [PostgresConnectorTask] hz.heuristic_neumann.jet.blocking.thread-0 - No previous offset found
07:03:37,656  INFO |ordersOfCustomers| - [InitialSnapshotter] hz.heuristic_neumann.jet.blocking.thread-0 - Taking initial snapshot for new datasource
07:03:37,661  INFO |ordersOfCustomers| - [PostgresReplicationConnection] hz.heuristic_neumann.jet.blocking.thread-0 - Creating replication slot with command CREATE_REPLICATION_SLOT "debezium"  LOGICAL decoderbufs
07:03:37,662  INFO |ordersOfCustomers| - [Threads] hz.heuristic_neumann.jet.blocking.thread-0 - Requested thread factory for connector PostgresConnector, id = 58fba57b-f9cf-49cf-9856-aafc50c65fc5 named = change-event-source-coordinator
07:03:37,662  INFO |ordersOfCustomers| - [Threads] hz.heuristic_neumann.jet.blocking.thread-0 - Creating thread debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator
07:03:37,663  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Metrics registered
07:03:37,663  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Context created
07:03:37,663  INFO |ordersOfCustomers| - [InitialSnapshotter] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Taking initial snapshot for new datasource
07:03:37,663  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - According to the connector configuration data will be snapshotted
07:03:37,663  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot step 1 - Preparing
07:03:37,663  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Setting isolation level
07:03:37,663  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Opening transaction with statement SET TRANSACTION ISOLATION LEVEL REPEATABLE READ; 
SET TRANSACTION SNAPSHOT '00000004-00000002-1';
07:03:37,680  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - The Kafka Connect schema name '58fba57b-f9cf-49cf-9856-aafc50c65fc5.inventory.customers.Value' is not a valid Avro schema name, so replacing with '_8fba57b_f9cf_49cf_9856_aafc50c65fc5.inventory.customers.Value'
07:03:37,680  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - The Kafka Connect schema name '58fba57b-f9cf-49cf-9856-aafc50c65fc5.inventory.customers.Key' is not a valid Avro schema name, so replacing with '_8fba57b_f9cf_49cf_9856_aafc50c65fc5.inventory.customers.Key'
07:03:37,680  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - The Kafka Connect schema name '58fba57b-f9cf-49cf-9856-aafc50c65fc5.inventory.customers.Envelope' is not a valid Avro schema name, so replacing with '_8fba57b_f9cf_49cf_9856_aafc50c65fc5.inventory.customers.Envelope'
07:03:37,680  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - The Kafka Connect schema name '58fba57b-f9cf-49cf-9856-aafc50c65fc5.inventory.orders.Value' is not a valid Avro schema name, so replacing with '_8fba57b_f9cf_49cf_9856_aafc50c65fc5.inventory.orders.Value'
07:03:37,680  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - The Kafka Connect schema name '58fba57b-f9cf-49cf-9856-aafc50c65fc5.inventory.orders.Key' is not a valid Avro schema name, so replacing with '_8fba57b_f9cf_49cf_9856_aafc50c65fc5.inventory.orders.Key'
07:03:37,680  WARN |ordersOfCustomers| - [SchemaNameAdjuster] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - The Kafka Connect schema name '58fba57b-f9cf-49cf-9856-aafc50c65fc5.inventory.orders.Envelope' is not a valid Avro schema name, so replacing with '_8fba57b_f9cf_49cf_9856_aafc50c65fc5.inventory.orders.Envelope'
07:03:37,680  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot step 2 - Determining captured tables
07:03:37,682  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Adding table inventory.geom to the list of capture schema tables
07:03:37,682  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Adding table inventory.products_on_hand to the list of capture schema tables
07:03:37,682  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Adding table inventory.customers to the list of capture schema tables
07:03:37,682  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Adding table inventory.orders to the list of capture schema tables
07:03:37,682  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Adding table inventory.products to the list of capture schema tables
07:03:37,682  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot step 3 - Locking captured tables [inventory.customers, inventory.orders]
07:03:37,682  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot step 4 - Determining snapshot offset
07:03:37,682  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Creating initial offset context
07:03:37,683  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Read xlogStart at 'LSN{0/2039DB8}' from transaction '596'
07:03:37,683  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Read xlogStart at 'LSN{0/2039DB8}' from transaction '596'
07:03:37,683  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot step 5 - Reading structure of captured tables
07:03:37,683  INFO |ordersOfCustomers| - [PostgresSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Reading structure of schema 'inventory' of catalog 'postgres'
07:03:37,705  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot step 6 - Persisting schema history
07:03:37,705  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot step 7 - Snapshotting data
07:03:37,706  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshotting contents of 2 tables while still in transaction
07:03:37,706  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Exporting data from table 'inventory.customers' (1 of 2 tables)
07:03:37,706  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - 	 For table 'inventory.customers' using select statement: 'SELECT "id", "first_name", "last_name", "email" FROM "inventory"."customers"'
07:03:37,709  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - 	 Finished exporting 4 records for table 'inventory.customers'; total duration '00:00:00.003'
07:03:37,709  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Exporting data from table 'inventory.orders' (2 of 2 tables)
07:03:37,710  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - 	 For table 'inventory.orders' using select statement: 'SELECT "id", "order_date", "purchaser", "quantity", "product_id" FROM "inventory"."orders"'
07:03:37,713  INFO |ordersOfCustomers| - [RelationalSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - 	 Finished exporting 4 records for table 'inventory.orders'; total duration '00:00:00.004'
07:03:37,713  INFO |ordersOfCustomers| - [AbstractSnapshotChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot - Final stage
07:03:37,714  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Snapshot ended with SnapshotResult [status=COMPLETED, offset=PostgresOffsetContext [sourceInfoSchema=Schema{io.debezium.connector.postgresql.Source:STRUCT}, sourceInfo=source_info[server='58fba57b-f9cf-49cf-9856-aafc50c65fc5'db='postgres', lsn=LSN{0/2039DB8}, txId=596, timestamp=2022-07-06T07:03:37.713Z, snapshot=FALSE, schema=inventory, table=orders], lastSnapshotRecord=true, lastCompletelyProcessedLsn=null, lastCommitLsn=null, streamingStoppingLsn=null, transactionContext=TransactionContext [currentTransactionId=null, perTableEventCount={}, totalEventCount=0], incrementalSnapshotContext=IncrementalSnapshotContext [windowOpened=false, chunkEndPosition=null, dataCollectionsToSnapshot=[], lastEventKeySent=null, maximumKey=null]]]
07:03:37,714  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Connected metrics set to 'true'
07:03:37,727  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.customers' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:03:37,727  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.orders' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:03:37,728  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Starting streaming
07:03:37,728  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Retrieved latest position from stored offset 'LSN{0/2039DB8}'
07:03:37,728  INFO |ordersOfCustomers| - [WalPositionLocator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Looking for WAL restart position for last commit LSN 'null' and last change LSN 'LSN{0/2039DB8}'
07:03:37,734  INFO |ordersOfCustomers| - [PostgresConnection] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{0/2039DB8}, catalogXmin=596]
07:03:37,735  INFO |ordersOfCustomers| - [JdbcConnection] pool-266-thread-1 - Connection gracefully closed
07:03:37,747  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Requested thread factory for connector PostgresConnector, id = 58fba57b-f9cf-49cf-9856-aafc50c65fc5 named = keep-alive
07:03:37,747  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Creating thread debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-keep-alive
07:03:37,761  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.customers' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:03:37,761  INFO |ordersOfCustomers| - [PostgresSchema] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - REPLICA IDENTITY for 'inventory.orders' is 'FULL'; UPDATE AND DELETE events will contain the previous values of all the columns
07:03:37,762  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Searching for WAL resume position
07:03:38,269  INFO |ordersOfCustomers| - [WalPositionLocator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - First LSN 'LSN{0/203DC28}' received
07:03:38,269  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - WAL resume position 'LSN{0/203DC28}' discovered
07:03:38,270  INFO |ordersOfCustomers| - [JdbcConnection] pool-267-thread-1 - Connection gracefully closed
07:03:38,285  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Requested thread factory for connector PostgresConnector, id = 58fba57b-f9cf-49cf-9856-aafc50c65fc5 named = keep-alive
07:03:38,285  INFO |ordersOfCustomers| - [Threads] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Creating thread debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-keep-alive
07:03:38,285  INFO |ordersOfCustomers| - [PostgresStreamingChangeEventSource] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Processing messages
07:03:38,285  INFO |ordersOfCustomers| - [WalPositionLocator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Message with LSN 'LSN{0/203DC28}' arrived, switching off the filtering
07:05:39,432  INFO || - [BaseSourceTask] ForkJoinPool.commonPool-worker-3 - Stopping down connector
07:05:39,537  INFO |ordersOfCustomers| - [JdbcConnection] pool-268-thread-1 - Connection gracefully closed
07:05:39,537  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Finished streaming
07:05:39,537  INFO |ordersOfCustomers| - [ChangeEventSourceCoordinator] debezium-postgresconnector-58fba57b-f9cf-49cf-9856-aafc50c65fc5-change-event-source-coordinator - Connected metrics set to 'false'
07:05:39,539  INFO || - [JdbcConnection] pool-269-thread-1 - Connection gracefully closed
07:05:39,541  INFO |ordersOfCustomers| - [MasterJobContext] hz.heuristic_neumann.cached.thread-13 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution of job '0848-d36e-e0c0-0001', execution 0848-d36e-e0c1-0001 got terminated, reason=java.util.concurrent.CancellationException
	Start time: 2022-07-06T07:03:37.614
	Duration: 00:02:01.927
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
07:05:39,543  INFO |ordersOfCustomers| - [JetTestSupport] Time-limited test - Terminating instanceFactory in JetTestSupport.@After
07:05:39,543  INFO |ordersOfCustomers| - [LifecycleService] Thread-112 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
07:05:39,543  WARN |ordersOfCustomers| - [Node] Thread-112 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Terminating forcefully...
07:05:39,543  INFO |ordersOfCustomers| - [Node] Thread-112 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down connection manager...
07:05:39,544  INFO |ordersOfCustomers| - [Node] Thread-112 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down node engine...
07:05:39,551  INFO |ordersOfCustomers| - [NodeExtension] Thread-112 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Destroying node NodeExtension.
07:05:39,551  INFO |ordersOfCustomers| - [Node] Thread-112 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 8 ms.
07:05:39,551  INFO |ordersOfCustomers| - [LifecycleService] Thread-112 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
07:05:39,968 DEBUG |ordersOfCustomers| - [ResourceReaper] Time-limited test - Removed container and associated volume(s): debezium/example-postgres:1.7
BuildInfo right after ordersOfCustomers(com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest): BuildInfo{version='5.2-SNAPSHOT', build='20220706', buildNumber=20220706, revision=be07ca9, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'ordersOfCustomers(com.hazelcast.jet.cdc.postgres.MultiTableCacheIntegrationTest):'
07:03:30, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:03:35, accumulated pauses: 307 ms, max pause: 255 ms, pauses over 1000 ms: 0
07:03:40, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:03:45, accumulated pauses: 49 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:03:50, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:03:55, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:00, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:05, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:10, accumulated pauses: 48 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:15, accumulated pauses: 43 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:20, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:25, accumulated pauses: 29 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:30, accumulated pauses: 28 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:35, accumulated pauses: 25 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:40, accumulated pauses: 41 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:45, accumulated pauses: 43 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:50, accumulated pauses: 42 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:04:55, accumulated pauses: 39 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:05:00, accumulated pauses: 47 ms, max pause: 1 ms, pauses over 1000 ms: 0
07:05:05, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:05:10, accumulated pauses: 46 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:05:15, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:05:20, accumulated pauses: 45 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:05:25, accumulated pauses: 44 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:05:30, accumulated pauses: 43 ms, max pause: 0 ms, pauses over 1000 ms: 0
07:05:35, accumulated pauses: 334 ms, max pause: 293 ms, pauses over 1000 ms: 0
```

</details>

--- a/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/RecordPartImpl.java
+++ b/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/RecordPartImpl.java
@@ -86,7 +86,7 @@ class RecordPartImpl implements RecordPart {
 
     @Override
     public int hashCode() {
-        return json.hashCode();
+        return toJson().hashCode();
     }
 
     @Override
@@ -98,7 +98,7 @@ class RecordPartImpl implements RecordPart {
             return false;
         }
         RecordPartImpl other = (RecordPartImpl) obj;
-        return Objects.equals(json, other.json);
+        return Objects.equals(toJson(), other.toJson());
     }
 
     @Override

--- a/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
+++ b/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
@@ -17,12 +17,8 @@
 package com.hazelcast.jet.cdc.postgres;
 
 import com.hazelcast.core.HazelcastInstance;
-import com.hazelcast.jet.Job;
 import com.hazelcast.jet.cdc.ChangeRecord;
-import com.hazelcast.jet.cdc.Operation;
 import com.hazelcast.jet.cdc.ParsingException;
-import com.hazelcast.jet.cdc.RecordPart;
-import com.hazelcast.jet.function.TriFunction;
 import com.hazelcast.jet.pipeline.Pipeline;
 import com.hazelcast.jet.pipeline.Sinks;
 import com.hazelcast.jet.pipeline.StreamSource;
@@ -42,18 +38,20 @@ import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Objects;
-import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
+import static com.hazelcast.jet.cdc.Operation.DELETE;
+import static com.hazelcast.jet.cdc.Operation.INSERT;
+import static com.hazelcast.jet.cdc.Operation.SYNC;
 import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+import static java.util.Objects.requireNonNull;
 
 @Category(NightlyTest.class)
 public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrationTest {
 
-    private static final int MAX_CONCURRENT_OPERATIONS = 1;
     private static final String CACHE = "cache";
-    private static final int REPEATS = 1000;
+    private static final int REPEATS = 1;
 
     @Test
     public void ordersOfCustomers() throws Exception {
@@ -63,25 +61,23 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
 
         Pipeline pipeline = Pipeline.create();
         StreamStage<ChangeRecord> allRecords = pipeline.readFrom(source)
-                .withNativeTimestamps(0);
+                .withNativeTimestamps(0).setLocalParallelism(1);
 
         allRecords.filter(r -> r.table().equals("customers"))
-                .apply(this::fixOrdering)
-                .writeTo(Sinks.mapWithEntryProcessor(MAX_CONCURRENT_OPERATIONS, CACHE,
+                .writeTo(Sinks.mapWithEntryProcessor(CACHE,
                         record -> (Integer) record.key().toMap().get("id"),
                         CustomerEntryProcessor::new
                 ));
 
         allRecords.filter(r -> r.table().equals("orders"))
-                .apply(this::fixOrdering)
-                .writeTo(Sinks.mapWithEntryProcessor(MAX_CONCURRENT_OPERATIONS, CACHE,
+                .writeTo(Sinks.mapWithEntryProcessor(CACHE,
                         record -> (Integer) record.value().toMap().get("purchaser"),
                         OrderEntryProcessor::new
                 ));
 
         // when
         HazelcastInstance hz = createHazelcastInstances(1)[0];
-        Job job = hz.getJet().newJob(pipeline);
+        hz.getJet().newJob(pipeline);
         //then
         Map<Integer, OrdersOfCustomer> expected = toMap(
         new OrdersOfCustomer(
@@ -96,22 +92,20 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
                 new Order(10004, new Date(1456012800000L), 1003, 1, 107)),
         new OrdersOfCustomer(
                 new Customer(1004, "Anne", "Kretchmar", "annek@noanswer.org")));
-        assertEqualsEventually(() -> getIMapContent(hz, CACHE), expected);
+        assertEqualsEventually(() -> getIMapContent(hz), expected);
 
         //when
         List<String> batch = new ArrayList<>();
+        batch.add("INSERT INTO customers VALUES (1005, 'Jason', 'Bourne', 'jason@bourne.org')");
         for (int i = 1; i <= REPEATS; i++) {
             batch.add("UPDATE customers SET first_name='Anne" + i + "' WHERE id=1004");
-
-            batch.add("INSERT INTO customers VALUES (1005, 'Jason', 'Bourne', 'jason@bourne.org')");
-            batch.add("DELETE FROM customers WHERE id=1005");
-
             batch.add("UPDATE orders SET quantity='" + i + "' WHERE id=10004");
 
             batch.add("DELETE FROM orders WHERE id=10003");
-            batch.add("INSERT INTO orders VALUES (10003, '2016-02-19', 1002, 2, 106)");
+            batch.add("INSERT INTO orders VALUES (10007, '2016-02-19', 1002, 2, 106)");
         }
         executeBatch(batch.toArray(new String[0]));
+        executeBatch("DELETE FROM customers WHERE id=1005");
 
         //then
         expected = toMap(
@@ -121,73 +115,25 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
                 new OrdersOfCustomer(
                         new Customer(1002, "George", "Bailey", "gbailey@foobar.com"),
                         new Order(10002, new Date(1452988800000L) , 1002, 2, 105),
-                        new Order(10003, new Date(1455840000000L), 1002, 2, 106)),
+                        new Order(10007, new Date(1455840000000L), 1002, 2, 106)),
                 new OrdersOfCustomer(
                         new Customer(1003, "Edward", "Walker", "ed@walker.com"),
                         new Order(10004, new Date(1456012800000L), 1003, REPEATS, 107)),
                 new OrdersOfCustomer(
                         new Customer(1004, "Anne" + REPEATS, "Kretchmar", "annek@noanswer.org")));
         expected.put(1005, new OrdersOfCustomer());
-        assertEqualsEventually(() -> getIMapContent(hz, CACHE), expected);
-    }
-
-    private StreamStage<ChangeRecord> fixOrdering(StreamStage<ChangeRecord> input) {
-        return input
-                .groupingKey(ChangeRecord::key)
-                .mapStateful(
-                        TimeUnit.SECONDS.toMillis(10),
-                        () -> new Sequence(0, 0),
-                        (lastSequence, key, record) -> {
-                            long source = record.sequenceSource();
-                            long sequence = record.sequenceValue();
-                            if (lastSequence.update(source, sequence)) {
-                                return record;
-                            }
-                            return null;
-                        },
-                        (TriFunction<Sequence, RecordPart, Long, ChangeRecord>) (sequence, recordPart, aLong) -> null);
+        assertEqualsEventually(() -> getIMapContent(hz), expected);
     }
 
     @Nonnull
-    private static Map<Integer, OrdersOfCustomer> getIMapContent(HazelcastInstance hz, String name) {
-        return new HashMap<>(hz.getMap(name));
+    private static Map<Integer, OrdersOfCustomer> getIMapContent(HazelcastInstance hz) {
+        return new HashMap<>(hz.getMap(MultiTableCacheIntegrationTest.CACHE));
     }
 
     @Nonnull
     private static Map<Integer, OrdersOfCustomer> toMap(OrdersOfCustomer... ordersOfCustomers) {
         return Arrays.stream(ordersOfCustomers).collect(Collectors.toMap(
                 orders -> orders.getCustomer().getId(), Function.identity()));
-    }
-
-    private static class Sequence {
-
-        private long source;
-        private long sequence;
-
-        Sequence(long source, long sequence) {
-            this.source = source;
-            this.sequence = sequence;
-        }
-
-        boolean update(long source, long sequence) {
-            if (this.source != source) { //sequence source changed for key
-                this.source = source;
-                this.sequence = sequence;
-                return true;
-            }
-
-            if (this.sequence < sequence) { //sequence is newer than previous for key
-                this.sequence = sequence;
-                return true;
-            }
-
-            return false;
-        }
-
-        @Override
-        public String toString() {
-            return "source=" + source + ", sequence=" + sequence;
-        }
     }
 
     private static class OrdersOfCustomer implements Serializable {
@@ -241,7 +187,7 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
 
         @Override
         public String toString() {
-            return String.format("Customer: %s, Orders: %s", customer, orders);
+            return String.format("%s, Orders: %s", customer, orders);
         }
     }
 
@@ -256,16 +202,14 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
         @Override
         public Object process(Entry<Integer, OrdersOfCustomer> entry) {
             try {
-                boolean deletion = Operation.DELETE.equals(record.operation());
                 OrdersOfCustomer value = entry.getValue();
-                if (deletion) {
-                    if (value != null) {
-                        value.setCustomer(null);
-                    }
+                if (value == null && (record.operation() == SYNC || record.operation() == INSERT)) {
+                    value = new OrdersOfCustomer();
+                }
+                requireNonNull(value, "value is null");
+                if (DELETE == record.operation()) {
+                    value.setCustomer(null);
                 } else {
-                    if (value == null) {
-                        value = new OrdersOfCustomer();
-                    }
                     value.setCustomer(record.value().toObject(Customer.class));
                 }
                 entry.setValue(value);
@@ -287,7 +231,7 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
         @Override
         public Object process(Entry<Integer, OrdersOfCustomer> entry) {
             try {
-                boolean deletion = Operation.DELETE.equals(record.operation());
+                boolean deletion = DELETE.equals(record.operation());
                 OrdersOfCustomer value = entry.getValue();
                 if (deletion) {
                     if (value != null) {

--- a/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
+++ b/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
@@ -45,13 +45,14 @@ import static com.hazelcast.jet.cdc.Operation.DELETE;
 import static com.hazelcast.jet.cdc.Operation.INSERT;
 import static com.hazelcast.jet.cdc.Operation.SYNC;
 import static com.hazelcast.jet.impl.util.ExceptionUtil.rethrow;
+import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
 @Category(NightlyTest.class)
 public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrationTest {
 
     private static final String CACHE = "cache";
-    private static final int REPEATS = 1;
+    private static final int REPEATS = 1000;
 
     @Test
     public void ordersOfCustomers() throws Exception {
@@ -102,10 +103,10 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
             batch.add("UPDATE orders SET quantity='" + i + "' WHERE id=10004");
 
             batch.add("DELETE FROM orders WHERE id=10003");
-            batch.add("INSERT INTO orders VALUES (10007, '2016-02-19', 1002, 2, 106)");
         }
         executeBatch(batch.toArray(new String[0]));
         executeBatch("DELETE FROM customers WHERE id=1005");
+        executeBatch("INSERT INTO orders VALUES (10007, '2016-02-19', 1002, 2, 106)");
 
         //then
         expected = toMap(
@@ -187,7 +188,7 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
 
         @Override
         public String toString() {
-            return String.format("%s, Orders: %s", customer, orders);
+            return format("%s, Orders: %s", customer, orders);
         }
     }
 

--- a/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
+++ b/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java
@@ -232,7 +232,7 @@ public class MultiTableCacheIntegrationTest extends AbstractPostgresCdcIntegrati
         @Override
         public Object process(Entry<Integer, OrdersOfCustomer> entry) {
             try {
-                boolean deletion = DELETE.equals(record.operation());
+                boolean deletion = DELETE == record.operation();
                 OrdersOfCustomer value = entry.getValue();
                 if (deletion) {
                     if (value != null) {

--- a/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/PostgresCdcAuthAndConnectionIntegrationTest.java
+++ b/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/PostgresCdcAuthAndConnectionIntegrationTest.java
@@ -36,7 +36,7 @@ public class PostgresCdcAuthAndConnectionIntegrationTest extends AbstractPostgre
     @Test
     public void wrongPassword() {
         StreamSource<ChangeRecord> source = PostgresCdcSources.postgres("name")
-                .setDatabaseAddress(postgres.getContainerIpAddress())
+                .setDatabaseAddress(postgres.getHost())
                 .setDatabasePort(postgres.getMappedPort(POSTGRESQL_PORT))
                 .setDatabaseUser("postgres")
                 .setDatabasePassword("wrongPassword")
@@ -51,14 +51,14 @@ public class PostgresCdcAuthAndConnectionIntegrationTest extends AbstractPostgre
         Job job = hz.getJet().newJob(pipeline);
         // then
         assertThatThrownBy(job::join)
-                .hasRootCauseInstanceOf(JetException.class)
+                .hasCauseInstanceOf(JetException.class)
                 .hasStackTraceContaining("password authentication failed for user \"postgres\"");
     }
 
     @Test
     public void incorrectDatabaseName() {
         StreamSource<ChangeRecord> source = PostgresCdcSources.postgres("name")
-                .setDatabaseAddress(postgres.getContainerIpAddress())
+                .setDatabaseAddress(postgres.getHost())
                 .setDatabasePort(postgres.getMappedPort(POSTGRESQL_PORT))
                 .setDatabaseUser("postgres")
                 .setDatabasePassword("postgres")
@@ -73,7 +73,7 @@ public class PostgresCdcAuthAndConnectionIntegrationTest extends AbstractPostgre
         Job job = hz.getJet().newJob(pipeline);
         // then
         assertThatThrownBy(job::join)
-                .hasRootCauseInstanceOf(JetException.class)
+                .hasCauseInstanceOf(JetException.class)
                 .hasStackTraceContaining("database \"wrongDatabaseName\" does not exist");
     }
 

--- a/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/PostgresCdcNetworkIntegrationTest.java
+++ b/extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/PostgresCdcNetworkIntegrationTest.java
@@ -43,6 +43,7 @@ import org.junit.runners.Parameterized.Parameters;
 import org.testcontainers.containers.Network;
 import org.testcontainers.containers.PostgreSQLContainer;
 import org.testcontainers.containers.ToxiproxyContainer;
+import org.testcontainers.utility.DockerImageName;
 
 import javax.annotation.Nonnull;
 import java.sql.Connection;
@@ -55,6 +56,7 @@ import java.util.function.Consumer;
 
 import static com.hazelcast.jet.Util.entry;
 import static com.hazelcast.jet.cdc.postgres.AbstractPostgresCdcIntegrationTest.getConnection;
+import static com.hazelcast.jet.cdc.postgres.PostgresCdcSources.PostgresSnapshotMode.INITIAL;
 import static com.hazelcast.jet.core.JobStatus.RUNNING;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
 import static java.util.concurrent.TimeUnit.SECONDS;
@@ -64,12 +66,16 @@ import static org.junit.Assert.assertTrue;
 import static org.junit.runners.Parameterized.UseParametersRunnerFactory;
 import static org.testcontainers.containers.PostgreSQLContainer.POSTGRESQL_PORT;
 
+@SuppressWarnings("resource")
 @RunWith(HazelcastParametrizedRunner.class)
 @UseParametersRunnerFactory(HazelcastSerialParametersRunnerFactory.class)
 @Category({NightlyTest.class})
 public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTest {
 
     private static final long RECONNECT_INTERVAL_MS = SECONDS.toMillis(1);
+    private static final DockerImageName TOXI_PROXY_IMAGE = DockerImageName
+            .parse("ghcr.io/shopify/toxiproxy:2.4.0")
+            .asCompatibleSubstituteFor("shopify/toxiproxy");
 
     @Parameter(value = 0)
     public RetryStrategy reconnectBehavior;
@@ -99,10 +105,10 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
     }
 
     @Test
-    public void when_noDatabaseToConnectTo() throws Exception {
+    public void when_noDatabaseToConnectTo() {
         postgres = initPostgres(null, null);
         int port = fixPortBinding(postgres, POSTGRESQL_PORT);
-        String containerIpAddress = postgres.getContainerIpAddress();
+        String containerIpAddress = postgres.getHost();
         stopContainer(postgres);
 
         Pipeline pipeline = initPipeline(containerIpAddress, port);
@@ -115,7 +121,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
         if (neverReconnect) {
             // then job fails
             assertThatThrownBy(job::join)
-                    .hasRootCauseInstanceOf(JetException.class)
+                    .hasCauseInstanceOf(JetException.class)
                     .hasStackTraceContaining("Failed to connect to database");
             assertTrue(hz.getMap("results").isEmpty());
         } else {
@@ -139,7 +145,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
     public void when_shortNetworkDisconnectDuringSnapshotting_then_connectorDoesNotNoticeAnything() throws Exception {
         try (
                 Network network = initNetwork();
-                ToxiproxyContainer toxiproxy = initToxiproxy(network);
+                ToxiproxyContainer toxiproxy = initToxiproxy(network)
         ) {
             postgres = initPostgres(network, null);
             ToxiproxyContainer.ContainerProxy proxy = initProxy(toxiproxy, postgres);
@@ -177,7 +183,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
         postgres = initPostgres(null, null);
         int port = fixPortBinding(postgres, POSTGRESQL_PORT);
 
-        Pipeline pipeline = initPipeline(postgres.getContainerIpAddress(), port);
+        Pipeline pipeline = initPipeline(postgres.getHost(), port);
         // when job starts
         HazelcastInstance hz = createHazelcastInstances(2)[0];
         Job job = hz.getJet().newJob(pipeline);
@@ -195,7 +201,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
         if (neverReconnect) {
             // then job fails
             assertThatThrownBy(job::join)
-                    .hasRootCauseInstanceOf(JetException.class)
+                    .hasCauseInstanceOf(JetException.class)
                     .hasStackTraceContaining("Failed to connect to database");
         } else {
             // and DB is started anew
@@ -215,7 +221,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
     public void when_shortConnectionLossDuringBinlogReading_then_connectorDoesNotNoticeAnything() throws Exception {
         try (
                 Network network = initNetwork();
-                ToxiproxyContainer toxiproxy = initToxiproxy(network);
+                ToxiproxyContainer toxiproxy = initToxiproxy(network)
         ) {
             postgres = initPostgres(network, null);
             ToxiproxyContainer.ContainerProxy proxy = initProxy(toxiproxy, postgres);
@@ -258,7 +264,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
         postgres = initPostgres(null, null);
         int port = fixPortBinding(postgres, POSTGRESQL_PORT);
 
-        Pipeline pipeline = initPipeline(postgres.getContainerIpAddress(), port);
+        Pipeline pipeline = initPipeline(postgres.getHost(), port);
         // when connector is up and transitions to binlog reading
         HazelcastInstance hz = createHazelcastInstances(2)[0];
         Job job = hz.getJet().newJob(pipeline);
@@ -274,7 +280,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
         if (neverReconnect) {
             // then job fails
             assertThatThrownBy(job::join)
-                    .hasRootCauseInstanceOf(JetException.class)
+                    .hasCauseInstanceOf(JetException.class)
                     .hasStackTraceContaining("Failed to connect to database");
         } else {
             // and results are cleared
@@ -310,6 +316,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
                 .setTableWhitelist("inventory.customers")
                 .setReconnectBehavior(reconnectBehavior)
                 .setShouldStateBeResetOnReconnect(resetStateOnReconnect)
+                .setSnapshotMode(INITIAL)
                 .build();
     }
 
@@ -331,6 +338,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
         }
     }
 
+    @SuppressWarnings("ConstantConditions")
     private PostgreSQLContainer<?> initPostgres(Network network, Integer fixedExposedPort) {
         PostgreSQLContainer<?> postgres = namedTestContainer(
                 new PostgreSQLContainer<>(AbstractPostgresCdcIntegrationTest.DOCKER_IMAGE)
@@ -339,7 +347,7 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
                         .withPassword("postgres")
         );
         if (fixedExposedPort != null) {
-            Consumer<CreateContainerCmd> cmd = e -> e.withPortBindings(
+            Consumer<CreateContainerCmd> cmd = e -> e.getHostConfig().withPortBindings(
                     new PortBinding(Ports.Binding.bindPort(fixedExposedPort), new ExposedPort(POSTGRESQL_PORT)));
             postgres = postgres.withCreateContainerCmdModifier(cmd);
         }
@@ -350,8 +358,9 @@ public class PostgresCdcNetworkIntegrationTest extends AbstractCdcIntegrationTes
         return postgres;
     }
 
+    @SuppressWarnings("resource")
     private ToxiproxyContainer initToxiproxy(Network network) {
-        ToxiproxyContainer toxiproxy = namedTestContainer(new ToxiproxyContainer().withNetwork(network));
+        ToxiproxyContainer toxiproxy = namedTestContainer(new ToxiproxyContainer(TOXI_PROXY_IMAGE).withNetwork(network));
         toxiproxy.start();
         return toxiproxy;
     }

